front/src/Phaser/Components/Ui/IconButton.ts

```diff
@@ -48,7 +49,17 @@ export class IconButton extends Phaser.GameObjects.Container {
         this.scene.add.existing(this);
     }
 
+    public disable(val = true): void {
+        this.disabled = val;
+        val ? this.disableInteractive() : this.setInteractive({ cursor: "pointer" });
+        this.background.setVisible(!val);
+        this.setAlpha(val ? 0.25 : 1);
+    }
```

Comment:
Am I correct to say this is not used and dead code?

Reply:
I would treat it more as a component feature, but yes.